### ISSUE TITLE
Error at install librabbitmq Big Sur 11.2.2

### ISSUE THREAD

**Issue by fjrd12** [celery/librabbitmq#157]
Env:

MacOS BigSur 11.2.2
Python 3.7.5
I encounter the below issue, any idea?
$ pip install librabbitmq

[error.txt](https://github.com/celery/librabbitmq/files/6134018/error.txt)

**Comment by thedrow** [comment]
See #127, #121.
Do any of the workarounds work for you?

**Comment by tombroi** [comment]
No, the above workarounds doesn't help. Fails on MacOS BigSur 11.2.3 too

**Comment by aichao** [comment]
Looking at the OP's `error.txt`, it appears that there are two compilation errors (same error, two places):

```
    .../Modules/_librabbitmq/connection.c:1359:13: error: implicit declaration of function 'amqp_simple_wait_frame_on_channel' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
    .../Modules/_librabbitmq/connection.c:1385:22: error: implicit declaration of function 'amqp_simple_wait_frame_on_channel' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
```

which suggests that `amqp_simple_wait_frame_on_channel` is not declared in `.../Modules/_librabbitmq/connection.c`. A quick grep of the source tree reveals that `amqp_simple_wait_frame_on_channel` is declared in `.../rabbitmq-c/librabbitmq/amqp_socket.h`. However, simply including `amqp_socket.h` is problematic because it redefines some data structures included from `amqp_tcp_socket.h`. This in itself is a bit smelly, which is probably the reason why we are here in the first place. I have no idea why this issue has not surfaced on other platforms other than the fact that configure may have simply set the flags to only give the warning instead of pedantically turn the warning into an error.

A simple, hackish workaround is to declare `amqp_simple_wait_frame_on_channel`, as declared in `amqp_socket.h`, at the top of `connection.c`. This is performed on the source of `librabbitmq` downloaded from [PyPI](http://pypi.python.org/pypi/librabbitmq). That is, after downloading and unpacking `librabbitmq-2.0.0` insert the following at the top of `librabbitmq-2.0.0/Modules/_librabbitmq/connection.c` (say at line 10 after `#include <amqp_tcp_socket.h>`):

```
    int
    amqp_simple_wait_frame_on_channel(amqp_connection_state_t state,
                                      amqp_channel_t channel,
                                      amqp_frame_t *decoded_frame);
```

Then, build and install from within the `librabbitmq-2.0.0` directory:

```
$ python setup.py build
$ python setup.py install
```

This worked for me on macOS 11.4 (Big Sur) with Xcode 12.4 and its Command line Tools and Python 3.9.5 (version of python should not matter as others have suggested in related issues). In addition, I had to upgrade `m4` to 1.4.19 and install the latest GNU autotools (`autoconf`, `automake`, and `libtool`) to configure and build properly (the powers-to-be in Cupertino are regressing to the Evil Empire in Redmond with every release of macOS). It appears that the OP did this already as his configure output appears clean.

Having said all this (sorry for being verbose), the real question for @thedrow and @ask is: what is the status of `librabbitmq`? From related issues it appears that it is abandoned and being replaced by `py-amqp`? From my original understanding, I thought `py-amqp` is a pure python implementation of a RabbitMQ client and is "slower" than `librabbitmq`. Has this been improved? @thedrow mentioned in #131 (April 2019):

> We're replacing this library with a custom implementation in Celery 5.

Is this now in Celery 5.1.1 itself (i.e., not a part of a bundled dependency)? Is this now the default when the broker is set to `amqp://`?

Thank you in advance for any clarifying answers.

**Comment by Chandan-Baba** [comment]
this issue is also surfacing on mac os Monterey. so, any help?

**Comment by sbernatsky** [comment]
For me following code has helped to solve "implicit declaration of function 'amqp_simple_wait_frame_on_channel'" erorr during librabbitmq installation:
```
export CFLAGS="-Wno-error=implicit-function-declaration"
pip install librabbitmq==2.0.0
```

This is for MacOS BigSur.

**Comment by icbd** [comment]
Install from source code:

```shell
git clone https://github.com/celery/librabbitmq.git
```

Edit `Makefile`:

```
# Distribuition tools
PYTHON=your-project-path/.venv/bin/python
```

> make install

**Comment by auvipy** [comment]
this is not the suggested package to use for a while. please use py-amqp
